sdk/storage/Azure.Storage.Files.Shares/src/Generated/FileRestClient.cs

```diff
@@ -12053,6 +12010,56 @@ public static partial class ShareModelFactory
 }
 #endregion class ShareSnapshotInfo
 
+#region enum ShareSnapshotsDeleteOption
+namespace Azure.Storage.Files.Shares.Models
+{
+    /// <summary>
+    /// Specifies the option include to delete the base share and all of its snapshots.
+    /// </summary>
+    public enum ShareSnapshotsDeleteOption
+    {
+        /// <summary>
+        /// include
+        /// </summary>
+        Include,
+
+        /// <summary>
+        /// include-leased
+        /// </summary>
+        IncludeLeased
+    }
```

Comment:
maybe `IncludeWithLeased` then or something like that.

Reply:
If we can add verbose docs strings without creating hand-written type then let's leave it. Otherwise (if adding docs would force type creation) let's rename.

Reply:
:/ I'd go with hand crafted type in that case. So that we can provide experience like this:
![image](https://user-images.githubusercontent.com/61715331/95876521-00e82980-0d28-11eb-9780-9b35d61ab7f8.png)

Reply:
Fixed.